File: testPublish.py
import paho.mqtt.publish as publish

#Test reading, change in microcontroller
pole_id = 2
waterlevel = 30

publish.single(
    f"sensors/{pole_id}/waterlevel",  # Topic, pole id is implicit from reading and waterlevel (int) is stored to the lowest level
    payload=str(waterlevel),          # Message payload, uploads waterlevel recorded to the lowest level topic
    hostname="10.244.80.97",             # Broker IP (database, change when accessed by microcontroller)
    auth={"username":"Sensor", "password":"Team13Capstone"},
    qos=1
)

File: ios-app/components/backend/testPublish.js
const mqtt = require("mqtt");

const client = mqtt.connect("mqtt://broker.hivemq.com");

client.on("connect", () => {
  console.log("Connected to test publisher. Sending continuous data...");

  const poleId = "Pole 1";
  let baseLevel = 45; // Starts in SAFE range

  // Send a new reading every 3 seconds
  setInterval(() => {
    // Randomly fluctuate between -5 and +15 to simulate gradually rising water
    const fluctuation = Math.floor(Math.random() * 20) - 5;
    baseLevel += fluctuation;
    
    // Clamp values between 0 and 100 for safety bounds
    if (baseLevel < 0) baseLevel = 0;
    if (baseLevel > 100) baseLevel = 100;

    const payload = JSON.stringify({ poleId, level: baseLevel });
    const topic = `safepass/sensors/${poleId}/waterlevel`;

    client.publish(topic, payload, { qos: 1 }, (err) => {
      if (err) {
        console.error("Failed to publish:", err);
      } else {
        console.log(`Published to ${topic}: ${payload}`);
      }
    });

  }, 3000); // 3000ms = 3 seconds
});

process.on("SIGINT", () => {
  console.log("\nStopping continuous test publisher...");
  client.end();
  process.exit();
});
